testsuite/MDAnalysisTests/analysis/test_contacts.py

```diff
@@ -171,16 +171,62 @@ class TestContacts(object):
     def universe():
         return mda.Universe(PSF, DCD)
 
-    def _run_Contacts(self, universe, start=None, stop=None, step=None, **kwargs):
+    def _run_Contacts(
+        self, universe,
+        ag_acidic=None, ag_basic=None,
+        start=None, stop=None, step=None, **kwargs
+    ):
         acidic = universe.select_atoms(self.sel_acidic)
         basic = universe.select_atoms(self.sel_basic)
+        if ag_acidic is None:
+            ag_acidic = self.sel_acidic
```

Comment:
I see what you're doing here, but I think it actually makes things harder to read, I would suggestion creating a new fixture that's just for dealing with atomgroups.

Reply:
Done, all tests are done on new static method `Contacts._get_atomgroup()`